src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -304,22 +304,19 @@ public bool DetermineExtensionDataProperty(Dictionary<string, JsonPropertyInfo>
             JsonPropertyInfo jsonPropertyInfo,
```

Comment:
Yes `matchingProperty` would be a super nice name, but I think the context is clear enough as is. Renaming this would be good to do when going through related code or doing some misc clean up.